extensions/amp-recaptcha-input/0.1/amp-recaptcha-service.js

```diff
@@ -193,6 +207,80 @@ export class AmpRecaptchaService {
     }
   }
 
+  /**
+   * Function to create our bootstrap iframe.
+   *
+   * @param {string} sitekey
+   * @return {!Promise<!Element>}
+   * @private
+   */
+  createRecaptchaFrame_(sitekey) {
+
+    const iframe = this.win_.document.createElement('iframe');
+
+    return this.getRecaptchaFrameSrc_().then(recaptchaFrameSrc => {
+      iframe.src = recaptchaFrameSrc;
+      iframe.ampLocation = parseUrlDeprecated(this.win_.location.href);
+      iframe.setAttribute('scrolling', 'no');
+      iframe.setAttribute('data-amp-3p-sentinel', 'amp-recaptcha');
+      iframe.setAttribute('name', JSON.stringify(dict({
+        'sitekey': sitekey,
+        'sentinel': 'amp-recaptcha',
+      })));
+      setStyle(iframe, 'border', 'none');
+      /** @this {!Element} */
+      iframe.onload = function() {
+        // Chrome does not reflect the iframe readystate.
+        this.readyState = 'complete';
+      };
+
+      return iframe;
+    });
+  }
+
+  /**
+   * Function to get our recaptcha iframe src
+   *
+   * This should take the current URL,
+   * either in canonical (www.example.com),
+   * or in cache (www-example-com.cdn.ampproject.org),
+   * and get the curls subdomain (www-example-com)
+   * To then create the iframe src:
+   * https://www-example-com.recaptcha.my.cdn/rtv/recaptcha.html
+   *
+   * @return {!Promise<string>}
+   * @private
+   */
+  getRecaptchaFrameSrc_() {
+    if (getMode().localDev || getMode().test) {
+      return tryResolve(() => {
+        return getDevelopmentBootstrapBaseUrl(this.win_, 'recaptcha');
+      });
+    }
+
+    // Need to have the curls subdomain match the original document url.
+    // This is verified by the recaptcha frame to
+    // verify the origin on its messages
+    let curlsSubdomainPromise = undefined;
+    if (isProxyOrigin(this.win_.location.href)) {
```

Comment:
Could use the viewerService's isProxyOrigin() method. But it would require you to install the service on doc level instead of window level. 

But please remind me why we want to service on window level?

Reply:
Discussed offline: Want to install on document level.

Get viewer service from ampdoc. 

(Do in a different PR)